internal/report/source_test.go

```diff
@@ -4,15 +4,58 @@ import (
 	"bytes"
 	"os"
 	"path/filepath"
+	"reflect"
 	"regexp"
 	"runtime"
 	"strings"
 	"testing"
 
 	"github.com/google/pprof/internal/binutils"
+	"github.com/google/pprof/internal/graph"
 	"github.com/google/pprof/profile"
 )
 
+func TestCalculatePercentiles(t *testing.T) {
+	for _, testCase := range []struct {
```

Comment:
I think there should be at least these additional test cases:
* Duplicate values.
* All equal values.